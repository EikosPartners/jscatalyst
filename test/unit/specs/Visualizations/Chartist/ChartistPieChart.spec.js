import Component from '@/components/visualizations/Chartist/ChartistPieChart'
import Vue from 'vue'
import { shallow, mount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import ChartistHelper from '@/common/chartistHelper.js'

const data = [ {"value": 24,"label": "Bubble"}, {"value": 22,"label": "Bar"}, {"value": 20,"label": "Line"}, {"value": 15,"label": "Scatter Plot"}, {"value": 30,"label": "Pie"}, {"value": 10,"label": "Area"}]

describe('Chartist Pie', () => {
    let wrapper
    beforeEach(() => {
      wrapper = mount(Component, {
          propsData: {
              dataModel: data
         },
         attachToDocument: true
      })
    })

    afterEach(() => {
        wrapper.destroy()
    })

    it('it renders at all', () => {
        expect(wrapper.contains('chartist')).toBe(true)
        expect(wrapper.html()).toContain('<chartist type="Pie"')
    })

    it('has the right prop data', () => {
      expect(wrapper.props().dataModel).toContainEqual({"value": 24,"label": "Bubble"})
      expect(wrapper.vm.propID).toBe('chartist-pie-chart')
    })

    it('has the right computed chart data', () => {
        expect(wrapper.vm.chartData).toHaveProperty('labels')
        expect(wrapper.vm.chartData).toHaveProperty('series')
    })


})

describe('Chartist Pie chart, snapshot', ()=>{
    let wrapper

    beforeEach(() => {
        wrapper = mount(Component, {
            propsData: {
                dataModel: data
           },
           attachToDocument: true
        })
    })

    it('matches snapshot', () => {
        let renderer = createRenderer()
        return Vue.nextTick()
            .then(function(){
                renderer.renderToString(wrapper.vm, (err, str) => {
                  if (err) throw new Error(err)
                  expect(str).toMatchSnapshot()
                })
            })

    })

})
