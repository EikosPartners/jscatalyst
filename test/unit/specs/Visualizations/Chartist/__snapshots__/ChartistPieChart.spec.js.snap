// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chartist Pie chart, snapshot matches snapshot 1`] = `
<div style="width:100%;height:100%;">
    <!---->
    <chartist type="Pie" data="[object Object]" options="[object Object]" responsiveOptions="screen and (min-width: 640px),[object Object]" id="chartist-pie-chart" style="height:100%;"></chartist>
</div>
`;
